Sources/EnumKit/Core/CaseAccessible.swift

```diff
@@ -144,16 +144,23 @@ public extension CaseAccessible {
 }
 
 private extension CaseAccessible {
-    func decompose<AssociatedValue>() -> (label: String, value: AssociatedValue)? {
+    func decompose<AssociatedValue>(expecting: AssociatedValue.Type) -> (path: [String?], value: AssociatedValue)? {
         let mirror = Mirror(reflecting: self)
         assert(mirror.displayStyle == .enum, "These CaseAccessible default functions should be used exclusively for enums")
-        guard mirror.displayStyle == .enum,
-            let pair = Mirror(reflecting: self).children.first,
-            let label = pair.label,
-            let result = (pair.value as? AssociatedValue) ??
-                (Mirror(reflecting: pair.value).children.first?.value as? AssociatedValue) else {
-                    return nil
+        guard mirror.displayStyle == .enum else { return nil }
+        
+        var path: [String?] = []
+        var any: Any = self
+        
+        while case let (label?, anyChild)? = Mirror(reflecting: any).children.first {
+            path.append(label)
+            path.append(String(describing: type(of: anyChild)))
+            if let child = anyChild as? AssociatedValue { return (path, child) }
+            any = anyChild
```

Comment:
This is pretty impressive. Took me some debugging to figure out how and why it works, but it does look like when the SR-12255 fix comes in this will kick in for empty associated values too.

Nice to see the ghost of single-element tuples is alive and well in the reflection system. 👻